### PR TITLE
fix(upsert): sets isNewRecord to false when upserting

### PR DESCRIPTION
**Pull request by knoid**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change
I'm having problems on upsert returned instances. Doing a `.save()` on an instance right after an upsert returns validation error because it tries to insert instead of updating. I think I tracked this error to having `isNewRecord` as `true` on that returned instance. I tried fixing it by setting it to false on `upsert` method but failed on other tests.

### CODE REVIEW

**Comment by sushantdhiman** [comment]
No travis build on this PR, can you try pushing an empty commit?

**Comment by knoid** [comment]
done! check Travis, it fails on MARIADB and I don't know how to fix that.

**Comment by knoid** [comment]
managed to fix all related errors! now it fails on mysql something totally unrelated. Am I in the right path? I needed to set `isNewRecord` to false manually after the insert because if not, it would not populate default values.

File: src/model.js [hunk 1]
```diff
@@ -2586,6 +2586,8 @@ class Model {
       options
     );
 
+    instance.isNewRecord = false;
```
Comment (sushantdhiman) [inline src/model.js:2589]: Although here `instance === result`, we should change naming here. May be rename `result` to `record` and then set `record.isNewRecord = false`

Comment (knoid) [reply src/model.js:2589]: Result is actually `[instance, boolean]`. Should I do this?
```suggestion
    const record = [result];
    record.isNewRecord = false;
```

Comment (sushantdhiman) [reply src/model.js:2589]: Yes, that works

Comment (sushantdhiman) [reply src/model.js]: Looks like tests are failing with this commit

**Comment by jingsam** [comment]
I encountered same problem. waiting to get this PR to be merged.

**Comment by sushantdhiman** [comment]
Please open a PR for `v6` branch to get a npm release

**Pull request merged**